internal/service/cluster/discovery/peer_discovery_test.go

```diff
@@ -0,0 +1,355 @@
+package discovery
+
+import (
+	"fmt"
+	stdlog "log"
+	"net"
+	"os"
+	"testing"
+
+	godiscover "github.com/hashicorp/go-discover"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/otel/trace/noop"
+
+	"github.com/go-kit/log"
+)
+
+func TestPeerDiscovery(t *testing.T) {
```

Comment:
These tests help to highlight some of the issues that I want to address on top of using A records in the follow-up PR.